src/js_printer.zig

```diff
@@ -1164,14 +1165,17 @@ fn NewPrinter(
         pub fn printBlock(p: *Printer, loc: logger.Loc, stmts: []const Stmt, close_brace_loc: ?logger.Loc) void {
             p.addSourceMapping(loc);
             p.print("{");
-            p.printNewline();
+            if (stmts.len > 0) {
+                @branchHint(.likely);
+                p.printNewline();
 
-            p.indent();
-            p.printBlockBody(stmts);
-            p.unindent();
-            p.needs_semicolon = false;
+                p.indent();
+                p.printBlockBody(stmts);
+                p.unindent();
+                p.needs_semicolon = false;
```

Comment:
`p.needs_semicolon = false;` should be unconditional right?

Reply:
correct. this can be moved outside since printIndent/addSourceMapping/print do not load this value